### PR TITLE
[MLA-1306] avoid copying png data

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Follow-up optimization attempt from https://github.com/Unity-Technologies/ml-agents/pull/4399

This was basically a wash performance-wise; we can keep if for cleanliness, or just drop the whole idea.

```
Tested with
mlagents-learn config/ppo/GridWorld.yaml --env=Project/Build/grid --force --num-envs=4
using 1/10 max steps

before
2020-08-26 20:29:32 INFO [stats.py:118] GridWorld. Step: 50000. Time Elapsed: 132.532 s. Mean Reward: -0.195. Std of Reward: 0.952. Training.

"process_pixels": {
    "total": 21.524973707002246,
    "count": 45987,
    "is_parallel": true,
    "self": 4.624946935002843,
    "children": {
        "image_decompress": {
            "total": 16.900026771999404,
            "count": 45987,
            "is_parallel": true,
            "self": 16.900026771999404
        }
    }
}


after
2020-08-26 20:44:20 INFO [stats.py:118] GridWorld. Step: 50000. Time Elapsed: 131.649 s. Mean Reward: -0.504. Std of Reward: 0.878. Training.

"process_pixels": {
    "total": 21.636028249000926,
    "count": 45711,
    "is_parallel": true,
    "self": 4.415242508999466,
    "children": {
        "image_decompress": {
            "total": 17.22078574000146,
            "count": 45711,
            "is_parallel": true,
            "self": 17.22078574000146
        }
    }
}
```

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [x] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents-envs/mlagents_envs/rpc_utils.py [hunk 1]
```diff
@@ -45,6 +45,35 @@ def behavior_spec_from_proto(
...
+            return res - self.offset
+        raise NotImplementedError()
+
+    def tell(self) -> int:
+        return self.fp.tell() - self.offset
+
+    def read(self, size: int = -1) -> bytes:
+        return self.fp.read(size)
+
+    def _real_tell(self) -> int:
```
Comment (harperj) [inline ml-agents-envs/mlagents_envs/rpc_utils.py:73]: Seems odd for this to be `_real_tell` when it's not used privately / internally.

Comment (chriselion) [reply ml-agents-envs/mlagents_envs/rpc_utils.py:73]: Yeah, wasn't sure what to call it. Do you think I should remove it and just use `fp.tell` directly?

Comment (harperj) [reply ml-agents-envs/mlagents_envs/rpc_utils.py:73]: Oh, I was just commenting on the initial underscore indicating it is a "private" method. I think "real_tell" is fine, or maybe "force_tell".

File: ml-agents-envs/mlagents_envs/rpc_utils.py [hunk 2]
```diff
@@ -70,7 +70,10 @@ def tell(self) -> int:
     def read(self, size: int = -1) -> bytes:
         return self.fp.read(size)
 
-    def _real_tell(self) -> int:
+    def original_tell(self) -> int:
```
Comment (harperj) [inline ml-agents-envs/mlagents_envs/rpc_utils.py:73]: 👍 even better

**Pull request merged**